diff --git a/Cpp/test/main.cc b/Cpp/test/main.cc
new file mode 100644
index 0000000..bbc2c29
--- /dev/null
+++ b/Cpp/test/main.cc
@@ -0,0 +1,25 @@
+#include <BabystepsTimer.h>
+
+#include "test.h"
+
+#include <thread>
+
+auto const scenarios = {
+  Scenario{
+    "BabystepsTimer is set to two minutes when created.",
+    []{
+      BabystepsTimer testee;
+
+      std::thread test_thread{[&testee]{
+        assert_that(ui_text(testee).find("02:00") != std::string::npos);
+        click(testee, "quit");
+      }};
+
+      testee.exec(0, nullptr);
+      test_thread.join();
+  }}
+};
+
+int main() {
+  return TestRunner().run(scenarios);
+}
